build

```diff
@@ -6,6 +6,5 @@ set -e
 if [[ "$1" ==  "--use-cache" ]]; then
     docker build --rm -t kaggle/python-build .
 else
-    docker pull continuumio/anaconda3:latest
-    docker build --rm --no-cache -t kaggle/python-build .
+    docker build --pull --rm --no-cache -t kaggle/python-build .
```

Comment:
Definitely better to have this, even though we use a pinned anaconda image, which I'm assuming to be immutable.  That is probably something we want to keep (the pinned part, not the specific version we use now), but the change here is less error-prone (in case we move away from a pin at some point).

Reply:
Docker tags ARE mutable. However, it is a best practice to treat them as immutable. I assume anaconda does, but we can't be sure.
On DTR, you can enforce tag immutability: https://docs.docker.com/ee/dtr/user/manage-images/prevent-tags-from-being-overwritten/

Reply:
Yeah I realize they are, just that in this case, given the version in them, we can expect immutability.   This change has the benefit of not assuming this, basically handling any mutability / floating refs, as said.

Didn't know about DTR, thanks for that ref.  Seems to enforce a best practice, and one we'd want to adopt.  I suppose it (or just this immutability enforcement) isn't available in `gcr.io`?

Reply:
I can't find a similar feature in `gcr.io`.